packages/desktop-client/src/hooks/useProperFocus.tsx

```diff
@@ -4,13 +4,16 @@ import React, {
   useLayoutEffect,
   useContext,
   useMemo,
+  type RefObject,
+  type ReactElement,
+  type MutableRefObject,
 } from 'react';
 
-function getFocusedKey(el) {
-  let node = el;
+function getFocusedKey(el: HTMLElement): string | null {
+  let node: HTMLElement | ParentNode = el;
   // Search up to 10 parent nodes
   for (let i = 0; i < 10 && node; i++) {
-    let key = node.dataset && node.dataset.focusKey;
+    let key = 'dataset' in node ? node.dataset?.focusKey : undefined;
```

Comment:
Small functional change here. But the logic should evaluate the same way as before.